src/governance-object.cpp

```diff
@@ -162,6 +162,21 @@ bool CGovernanceObject::ProcessVote(CNode* pfrom,
         LogPrint("gobject", "%s\n", ostr.str());
         exception = CGovernanceException(ostr.str(), GOVERNANCE_EXCEPTION_NONE);
         return false;
+    } else if (vote.GetTimestamp() == voteInstanceRef.nCreationTime) {
+        // Someone is doing smth fishy, there can be no two votes from the same masternode
+        // with the same timestamp for the same object and signal and yet different hash/outcome.
+        std::ostringstream ostr;
+        ostr << "CGovernanceObject::ProcessVote -- Invalid vote, same timestamp for the different outcome";
+        if (vote.GetOutcome() < voteInstanceRef.eOutcome) {
```

Comment:
Instead of choosing one, why not invalidate both votes if they are voting in a fishy way?

Reply:
Hmmm... in this case I think we would have to always relay this invalid vote further to make sure everyone is going to invalidate both of them too. And we would probably have to relay both (all?) these duplicates to other nodes when they try to sync to make sure they are on the same page to which means that we would have to store such duplicates in some new data structure because we normally don't relay invalid votes (stored in `cmapInvalidVotes`). With current solution we either relay this vote as usual (and it's going to replace the old one) or we don't relay it at all (because it's not a "winning" one and is going to be rejected by other nodes too), and also we relay only single "best" vote to nodes that are trying to sync.

Reply:
Great point, had not considered that at all.